File: c#/Services/DataProcessorService.cs
using LeagueRanking.Helpers;
using LeagueRanking.Interfaces;
using LeagueRanking.Models;

namespace LeagueRanking.Services
{
    /// <summary>
    /// This class processes the data from either the command line or file input. It implements IDataProcessor interface
    /// </summary>
    internal class DataProcessorService : IDataProcessor
    {
        
        List<Models.ResultModel.MatchResult> matchResults = new();

        /// <summary>
        /// Method to process a file
        /// </summary>
        /// <param name="file"></param>
        /// <returns></returns>
        public List<ResultModel.MatchResult> ProcessFile(string file)
        {
            string[] lines = File.ReadAllLines(file);

            string joined = string.Join(Environment.NewLine, lines);

            return ProcessResult(joined);
        }

        /// <summary>
        /// Method to process scores on the command line
        /// </summary>
        /// <param name="result"></param>
        /// <returns></returns>
        public List<ResultModel.MatchResult> ProcessResult(string result)
        {
            //Check if we have more than 1 result
            if (result.Contains(Environment.NewLine))
            {
                string[] resultPairs = result.Split(Environment.NewLine);

                for (int i = 0; i < resultPairs.Length; i++)
                    AddMatchResult(resultPairs[i]);
            }
            else AddMatchResult(result); //Only one result

            return matchResults;
        }

        /// <summary>
        /// Private method to add the match result to the match results list to be used later
        /// </summary>
        /// <param name="result"></param>
        private void AddMatchResult(string result)
        {
            //As we have 2 teams we split by comma
            string[] splitString = result.Split(',');

            //Add info for each team
            Models.ResultModel.TeamResult Team1 = new()
            {
                TeamName = ResultProcessorHelper.GetTeamName(splitString[0]),
                Score = ResultProcessorHelper.GetScore(splitString[0])
            };

            Models.ResultModel.TeamResult Team2 = new()
            {
                TeamName = ResultProcessorHelper.GetTeamName(splitString[1]),
                Score = ResultProcessorHelper.GetScore(splitString[1])
            };


            Team2.TeamName = ResultProcessorHelper.GetTeamName(splitString[1]);
            Team2.Score = ResultProcessorHelper.GetScore(splitString[1]);

            //Add to the matchresult list to be used later
            Models.ResultModel.MatchResult matchResult = new()
            {
                Team1 = Team1,
                Team2 = Team2
            };

            matchResults.Add(matchResult);
        }

    }
}

File: python/leagueranking/services/DataProcessorService.py
from leagueranking.models.resultmodel import match_result
from leagueranking.models.resultmodel import team_result
from leagueranking.helpers.resultprocessorhelper import ResultProcessorHelper
from leagueranking.helpers.logginghelper import LoggingHelper

# Service to process the data from the command line
class DataProcessorService:
    _match_results: match_result
    _results = []

    def __init__(self, results: str):
        self._match_results = []
        self._results = results

    # Main function to process the results
    def process_result(self):
        for result in self._results: # iterate through all the command line scores and add them to the match_results object
            try:
                self.add_match_result(result.strip())
            except Exception as Argument:
                LoggingHelper.log_console("An error occured trying to add the match result")
                LoggingHelper.log_file(Argument)

        return self._match_results

    # Function to add all the match results
    def add_match_result(self, result: str):
        # Split into team name and score
        scores = result.split(',')

        # Add team 1 results
        team1 = team_result()
        team1.team_name = ResultProcessorHelper.get_team_name(scores[0]) #call to helpers.resultprocessorhelper.py
        team1.score = ResultProcessorHelper.get_score(scores[0]) #call to helpers.resultprocessorhelper.py

        # Add team 2 results
        team2 = team_result()
        team2.team_name = ResultProcessorHelper.get_team_name((scores[1])) #call to helpers.resultprocessorhelper.py
        team2.score = ResultProcessorHelper.get_score(scores[1]) #call to helpers.resultprocessorhelper.py

        result = match_result()
        result.team1 = team1
        result.team2 = team2

        # Add to the match_results object to be used to process the rankings
        self._match_results.append(result)

